include/swift/ABI/MetadataValues.h

```diff
@@ -606,6 +615,55 @@ class TargetFunctionTypeFlags {
 };
 using FunctionTypeFlags = TargetFunctionTypeFlags<size_t>;
 
+template <typename int_type> class TargetParameterTypeFlags {
```

Comment:
We seem to be assuming that the backing storage for this is always a uint32_t.  I think it's still maybe a good idea to have it be templated, because it solves a lot of annoying calling-convention portability concerns to always pass a parameter as a full word instead of a smaller integer, but I think it merits a comment about why we're doing it.